### DISCUSSION TITLE
Can character controllers add physical materials?

### DISCUSSION THREAD

**Discussion by dannisliang** [bepu/bepuphysics1#39] [Q&A]
Can character controllers add physical materials?

**Comment by RossNordby** [comment]
Not sure what you mean. You can change the character's movement properties like `CharacterController.TractionForce` based on information like the `CharacterController.SupportFinder.SupportData`, if that's what you're looking for.

**Reply by dannisliang** [reply]
Want to change its dynamic or static friction. changed CharacterController.Body. Material. KineticFriction=10 and CharacterController.Body. Material. StaticFriction=10 seems to be ineffective

**Reply by RossNordby** [reply]
Character friction with the environment is explicitly removed because it would interfere with movement.

You could modify the `CharacterController.RemoveFriction` function to take into account contact normal or something like that to choose to remove friction only from a subset of pairs.

**Reply by dannisliang** [reply]
I really appreciate your help
